File: scripts/import.lua
local function read_path(prompt)
    io.write(prompt)
    io.flush()
    return io.read("*l")
end

local function shell_quote(path)
    return "'" .. path:gsub("'", "'\\''") .. "'"
end

local source = read_path("where should files be imported from? ")
local destination = read_path("where should the imported files go? ")

if not source or source == "" or not destination or destination == "" then
    io.stderr:write("source and destination are required.\n")
    os.exit(1)
end

local command = "mv -- " .. shell_quote(source) .. " " .. shell_quote(destination)
local success = os.execute(command)

if success ~= true and success ~= 0 then
    io.stderr:write("could not move the imported files.\n")
    os.exit(1)
end

io.write("imported files moved to " .. destination .. ".\n")

File: scripts/import.py
#!/usr/bin/env python3

import shutil
import sys
from pathlib import Path


def read_path(prompt: str) -> str:
    return input(prompt).strip()


def main() -> None:
    source = read_path("where should files be imported from? ")
    destination = read_path("where should the imported files go? ")

    if not source or not destination:
        sys.stderr.write("source and destination are required.\n")
        raise SystemExit(1)

    try:
        shutil.move(source, destination)
    except Exception:
        sys.stderr.write("could not move the imported files.\n")
        raise SystemExit(1)

    sys.stdout.write(f"imported files moved to {destination}.\n")


if __name__ == "__main__":
    main()
